cloudpickle/cloudpickle_fast.py

```diff
@@ -604,6 +590,30 @@ def __init__(self, file, protocol=None, buffer_callback=None):
             self.globals_ref = {}
             self.proto = int(protocol)
 
+            # Instance-level public dispatch_table should not mutate the
+            # private class-level _dispatch_table while still reflecting.
+            # Note that it is important to lazy init the dispatch_table
+            # public attribute to make it possible to re-assign it later
+            # and get the C implementation of Pickler take the
+            # re-assignment into account.
+            self.dispatch_table = ChainMap(
+                {},  # to register instance-level custom reducers
```

Comment:
Should we have an internal variable to point to this?

Reply:
That's a good suggestion. I still need to find the time to understand the cause of the PyPy segfault.